components/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt/src/main/java/org/wso2/carbon/identity/api/resource/mgt/dao/impl/APIResourceManagementDAOImpl.java

```diff
@@ -488,6 +490,51 @@ public List<ApplicationBasicInfo> getSubscribedApplications(String apiId) {
         return null;
     }
 
+    @Override
+    public List<APIResource> getScopeMetadata(List<String> scopeNames, Integer tenantId)
+            throws APIResourceMgtException {
+
+        String query = SQLConstants.GET_SCOPE_METADATA;
+        String placeholders = String.join(",", Collections.nCopies(scopeNames.size(), "?"));
+        query = query.replace(SQLConstants.SCOPE_LIST_PLACEHOLDER, placeholders);
+        try (Connection dbConnection = IdentityDatabaseUtil.getDBConnection(false)) {
+            PreparedStatement prepStmt = dbConnection.prepareStatement(query);
```

Comment:
include the preparted statement also as a resource in the same try block

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/5003/commits/ea4b23a67c37fdf0e9b8c39e8ee2dccd0167fbda